modules/widget/journal_entry_detail/JournalEntryDetailWidgetModule.php

```diff
@@ -85,6 +85,8 @@ public function addJournalEntryImage($iDocumentId) {
 		$oJournalEntryImage = new JournalEntryImage();
 		$oJournalEntryImage->setJournalEntryId($this->iJournalEntryId);
 		$oJournalEntryImage->setDocumentId($iDocumentId);
+		$iImagesCount= JournalEntryImageQuery::create()->filterByJournalEntryId($this->iJournalEntryId)->count();
+		$oJournalEntryImage->setSort($iImagesCount+1);
```

Comment:
Maybe set it to the highest value of any currently-existing image + 1 instead of just count + 1. This way you can more easily deal with deletions.

Reply:
commit 7428e48b026d5650683e9509e436fc0b5694e2d8